gograce.go

```diff
@@ -10,24 +10,41 @@ import (
 	"time"
 )
 
-func NewServerWithTimeout(t time.Duration) (*http.Server, chan struct{}) {
+var testQuit chan os.Signal
```

Comment:
Instead of this hack you could do something like this
https://github.com/xonvanetta/shutdown/blob/master/pkg/shutdown/shutdown_test.go#L38-L44

Reply:
I found this that i could be sent to the wrong thread so i did not want to use that. 
https://github.com/golang/go/issues/19326